pkg/controllers/rollout/override.go

```diff
@@ -0,0 +1,195 @@
+			if !r.InformerManager.IsClusterScopedResources(uResource.GroupVersionKind()) {
+				croKey := fleetv1beta1.ResourceIdentifier{
+					Group:   utils.NamespaceMetaGVK.Group,
+					Version: utils.NamespaceMetaGVK.Version,
+					Kind:    utils.NamespaceMetaGVK.Kind,
+					Name:    uResource.GetNamespace(),
+				}
+				possibleCROs[croKey] = true // selected by the namespace
+				roKey := fleetv1beta1.ResourceIdentifier{
+					Group:     uResource.GetObjectKind().GroupVersionKind().Group,
+					Version:   uResource.GetObjectKind().GroupVersionKind().Version,
+					Kind:      uResource.GetObjectKind().GroupVersionKind().Kind,
+					Namespace: uResource.GetNamespace(),
+					Name:      uResource.GetName(),
+				}
+				possibleROs[roKey] = true // selected by the object itself
+			} else {
+				croKey := fleetv1beta1.ResourceIdentifier{
+					Group:   uResource.GetObjectKind().GroupVersionKind().Group,
+					Version: uResource.GetObjectKind().GroupVersionKind().Version,
+					Kind:    uResource.GetObjectKind().GroupVersionKind().Kind,
+					Name:    uResource.GetName(),
+				}
+				possibleCROs[croKey] = true // selected by the object itself
+			}
+		}
+	}
+
+	croList := &fleetv1alpha1.ClusterResourceOverrideList{}
+	if err := r.Client.List(ctx, croList); err != nil {
+		klog.ErrorS(err, "Failed to list all the clusterResourceOverrides")
+		return nil, nil, err
+	}
+	roList := &fleetv1alpha1.ResourceOverrideList{}
+	if err := r.Client.List(ctx, roList); err != nil {
+		klog.ErrorS(err, "Failed to list all the resourceOverrides")
+		return nil, nil, err
+	}
+
+	filteredCRO := make([]*fleetv1alpha1.ClusterResourceOverride, 0, len(croList.Items))
+	filteredRO := make([]*fleetv1alpha1.ResourceOverride, 0, len(roList.Items))
+	for i := range croList.Items {
+		for _, selector := range croList.Items[i].Spec.ClusterResourceSelectors {
+			croKey := fleetv1beta1.ResourceIdentifier{
+				Group:   selector.Group,
+				Version: selector.Version,
+				Kind:    selector.Kind,
+				Name:    selector.Name,
+			}
+			if possibleCROs[croKey] {
+				filteredCRO = append(filteredCRO, &croList.Items[i])
+				break
+			}
+		}
+	}
+	for i := range roList.Items {
+		for _, selector := range roList.Items[i].Spec.ResourceSelectors {
+			roKey := fleetv1beta1.ResourceIdentifier{
+				Group:     selector.Group,
+				Version:   selector.Version,
+				Kind:      selector.Kind,
+				Namespace: roList.Items[i].Namespace,
+				Name:      selector.Name,
+			}
+			if possibleROs[roKey] {
+				filteredRO = append(filteredRO, &roList.Items[i])
+				break
+			}
+		}
+	}
+	return filteredCRO, filteredRO, nil
+}
+
+// getLatestOverridesPerBinding will look for any overrides associated with the "Bound" binding.
+func (r *Reconciler) getLatestOverridesPerBinding(ctx context.Context, binding *fleetv1beta1.ClusterResourceBinding, croList []*fleetv1alpha1.ClusterResourceOverride, roList []*fleetv1alpha1.ResourceOverride) ([]string, []fleetv1beta1.NamespacedName, error) {
```

Comment:
nit: the function name is more like pickFromResourceMatchedOverrideThatMatchesTargetCluster. Maybe add a few more comments?